internal/pkg/cachecli/cache_clean.go

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cachecli
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sylabs/singularity/internal/pkg/client/cache"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+// CleanLibraryCache : clean only library cache, will return a error if one occurs
+func CleanLibraryCache() error {
+	sylog.Debugf("Removing: %v", cache.Library())
+
+	err := os.RemoveAll(cache.Library())
+	if err != nil {
+		sylog.Warningf("Unable to clean cache: %v", err)
+	}
+
+	return err
+}
+
+// CleanOciCache : clean only oci cache, will return a error if one occurs
+func CleanOciCache() error {
+	sylog.Debugf("Removing: %v", cache.OciTemp())
+
+	err := os.RemoveAll(cache.OciTemp())
+	if err != nil {
+		sylog.Warningf("Unable to clean cache: %v", err)
+	}
+
+	return err
+}
+
+// CleanBlobCache : clean only blob cache, will return a error if one occurs
+func CleanBlobCache() error {
+	sylog.Debugf("Removing: %v", cache.OciBlob())
+
+	err := os.RemoveAll(cache.OciBlob())
+	if err != nil {
+		sylog.Warningf("Unable to clean cache: %v", err)
+	}
+
+	return err
+
+}
+
+func cleanLibraryCache(cacheName string) (bool, error) {
+	foundMatch := false
+	libraryCacheFiles, err := ioutil.ReadDir(cache.Library())
+	if err != nil {
+		sylog.Warningf("Unable to opening cache folder: %v", err)
+		return false, err
+	}
+	for _, f := range libraryCacheFiles {
+		cont, err := ioutil.ReadDir(filepath.Join(cache.Library(), f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to look in cache folder: %v", err)
+			return false, err
+		}
+		for _, c := range cont {
+			if c.Name() == cacheName {
+				sylog.Debugf("Removing: %v", filepath.Join(cache.Library(), f.Name(), c.Name()))
+				err = os.RemoveAll(filepath.Join(cache.Library(), f.Name(), c.Name()))
+				if err != nil {
+					sylog.Warningf("Unable to remove cache: %v", err)
+					return false, err
+				}
+				foundMatch = true
+			}
+		}
+	}
+
+	return foundMatch, nil
+}
+
+func cleanOciCache(cacheName string) (bool, error) {
+	foundMatch := false
+	blobs, err := ioutil.ReadDir(cache.OciTemp())
+	if err != nil {
+		sylog.Warningf("Unable to opening cache folder: %v", err)
+		return false, err
+	}
+	for _, f := range blobs {
+		blob, err := ioutil.ReadDir(filepath.Join(cache.OciTemp(), f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to look in cache folder: %v", err)
+			return false, err
+		}
+		for _, b := range blob {
+			if b.Name() == cacheName {
+				sylog.Debugf("Removing: %v", filepath.Join(cache.OciTemp(), f.Name(), b.Name()))
+				err = os.RemoveAll(filepath.Join(cache.OciTemp(), f.Name(), b.Name()))
```

Comment:
Here I think we need to make sure to remove the actual folder, not just the contents of it. So for example, I think we need to do `err = os.RemoveAll(filepath.Join(cache.OciTemp(), f.Name()))`, rather than only removing the block from within `f`.